packages/amplify-category-api/src/provider-utils/awscloudformation/utils/rds-resources/utils.ts

```diff
@@ -50,6 +54,13 @@ export const checkForUnsupportedDirectives = (schema: string, context: DataSourc
   visit(document, schemaVisitor);
 };
 
+export const containsSqlModelOrDirective = (dataSourceStrategies: Record<string, ModelDataSourceStrategy>): boolean => {
+  if (dataSourceStrategies && Object.keys(dataSourceStrategies).length > 0) {
```

Comment:
At this point are the `@sql` operations already added to the data source strategies, in an earlier pre-processing step?